runtime/drivers/duckdb/connectors.go

```diff
@@ -94,3 +105,17 @@ func getSourceReader(path string) (string, error) {
 		return "", fmt.Errorf("file type not supported : %s", ext)
 	}
 }
+
+func getMultiSourceReader(paths []string) (string, error) {
+	ext := fileutil.FullExt(paths[0])
+	dir := filepath.Dir(paths[0])
+	if ext == "" {
+		return "", fmt.Errorf("invalid file")
+	} else if strings.Contains(ext, ".csv") || strings.Contains(ext, ".tsv") || strings.Contains(ext, ".txt") {
+		return fmt.Sprintf("read_csv_auto('%s/*%s')", dir, ext), nil
+	} else if strings.Contains(ext, ".parquet") {
+		return fmt.Sprintf("read_parquet('%s/*%s')", dir, ext), nil
+	} else {
+		return "", fmt.Errorf("file type not supported : %s", ext)
+	}
+}
```

Comment:
1. The reason I am reading all files from a directory is because passing the entire list of files in a sql query can potentially reach the query limit if any(haven't been able to find any in the docs although some sql systems have limits on the number of characters in a query). 
That's the reason I am ingesting the files in a separate temp directory per ingestion and passing the directory itself to duck db. 
I agree to your point that inferring dir path from filename can be fragile. We can think of passing the directory name itself from the connector if required. 
Please let me know your thoughts.

Reply:
2. In case we continue to load all data form directory we will need getSourceReader for local ingestion

Reply:
I tested and there doesn't appear to be a query limit. So let's do it the "right" way and pass each file separately. Here's the test script I used:

```python
import duckdb
import os.path

n = 100000
paths = []
for i in range (0, n):
    name = f"data/data_{i}.csv"
    f = open(name, "w")
    f.write(f"a,b\n{i},{i}\n")
    f.close()
    paths.append(os.path.join(os.getcwd(), name))


sql = "create table foo as select * from "
sql += "read_csv_auto(['"
sql += "','".join(paths)
sql += "'])"

con = duckdb.connect(database=':memory:')
con.execute(sql)
con.execute("select * from foo limit 20")
print(con.fetchall())
```